### PR TITLE
dbrx

### PR DESCRIPTION
**Pull request by eitanturok**
# What does this PR do?

Add dbrx to `transformers` from https://github.com/databricks/dbrx/tree/main/model.

To run dbrx:
```python
from transformers import DbrxForCausalLM
model = DbrxForCausalLM.from_pretrained(
    "databricks/dbrx-instruct",
    device_map="auto",
    )
```

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by milocress** [comment]
GH Issue here: https://github.com/huggingface/transformers/issues/29911

**Comment by winglian** [comment]
you need to add dbrx to the `tokenization_auto.py` and `configuration_auto.py`

File: src/transformers/models/dbrx/tokenizer_config.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "add_prefix_space": false,
+  "bos_token": "<|endoftext|>",
+  "clean_up_tokenization_spaces": false,
+  "eos_token": "<|endoftext|>",
+  "model_max_length": 8192,
+  "tokenizer_class": "GPT4Tokenizer",
```
Comment (winglian) [inline src/transformers/models/dbrx/tokenizer_config.json:7]: There is no `GPT4Tokenizer` class already in transformers or this PR?

**Comment by eitanturok** [comment]
Closing this PR because it is being complete here #29921

**Pull request closed without merging**